subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -288,23 +293,43 @@ public void setDistributionPath(String distributionPath) {
         this.distributionPath = distributionPath;
     }
 
+    /**
+     * Set Wrapper versions resources.
+     *
+     * @since 8.1
+     */
+    @Incubating
+    public void setWrapperVersionsResources(WrapperVersionsResources wrapperVersionsResources) {
+        DefaultWrapperVersionsResources defaultWrapperVersionsResources = (DefaultWrapperVersionsResources) wrapperVersionsResources;
+        gradleVersionResolver.setTextResources(defaultWrapperVersionsResources.getLatest(),
+            defaultWrapperVersionsResources.getReleaseCandidate(),
+            defaultWrapperVersionsResources.getNightly(),
+            defaultWrapperVersionsResources.getReleaseNightly());
+    }
+
     /**
      * Returns the gradle version for the wrapper.
+     * This may throw if the label that can be provided via #setGradleVersion(String) can not be resolved at the moment
+     * e.g. there is not a `release-candidate` available at all times.
```

Comment:
:x: These two lines are now redundant with the `@throws` tag and should be removed.